lib/associations/mixin.js

```diff
@@ -209,18 +209,27 @@ const Mixin = {
     return association;
   },
 
-  getAssociation(target, alias) {
-    for (const associationName in this.associations) {
-      if (this.associations.hasOwnProperty(associationName)) {
-        const association = this.associations[associationName];
+  getAssociations(target) {
+    return Object.values(this.associations).filter(association => association.target.name === target.name);
+  },
 
-        if (association.target.name === target.name && (alias === undefined ? !association.isAliased : association.as === alias)) {
-          return association;
-        }
-      }
+  getAssociationForAlias(target, alias) {
```

Comment:
I added `getAssociations` because we need to know how many associations two models have in order to throw the most informative errors possible.

'getAssociationForAlias' is functionally identical to what used to be `getAssociation`. I renamed it to avoid confusion between it `getAssociations`, seeing as they are functionally get quite different now. I wanted the name to convey that it checks the association's alias as well.

`verifyAssociationAlias` is just a helper method